<h1>Activity 1</h1>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')
from scipy import stats
penguins = pd.read_csv('penguins.csv')
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [2]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [3]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [4]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [5]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [6]:
penguspecies = penguins['species'].value_counts()
penguspecies

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [7]:
Adelie_Pengu = penguins[penguins['species'] == 'Adelie']
Gentoo_Pengu = penguins[penguins['species'] == 'Gentoo']
Chinstrap_Pengu = penguins[penguins['species'] == 'Chinstrap']

In [8]:
Chinstrap_Pengu

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [9]:
fit_chinstrap = penguins[(penguins['species'] == 'Chinstrap') & (Chinstrap_Pengu["body_mass_g"] >= 4000)]
print(fit_chinstrap)

       species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream    

<h1>FILLUP NULL VALUES</h1>

In [10]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [11]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [12]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [13]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [14]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [15]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [16]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [17]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [18]:
penguins[penguins['island'] != 'Dream' ]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen        39.10000       18.70000         181.000000   
1    Adelie  Torgersen        39.50000       17.40000         186.000000   
2    Adelie  Torgersen        40.30000       18.00000         195.000000   
3    Adelie  Torgersen        43.92193       17.15117         200.915205   
4    Adelie  Torgersen        36.70000       19.30000         193.000000   
..      ...        ...             ...            ...                ...   
271  Gentoo     Biscoe        43.92193       17.15117         200.915205   
272  Gentoo     Biscoe        46.80000       14.30000         215.000000   
273  Gentoo     Biscoe        50.40000       15.70000         222.000000   
274  Gentoo     Biscoe        45.20000       14.80000         212.000000   
275  Gentoo     Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
271  4201.754386    male  2009  
272  4850.000000  female  2009  
273  5750.000000    male  2009  
274  5200.000000  female  2009  
275  5400.000000    male  2009  

[220 rows x 8 columns]

<h1>CHANGING DATA TYPE OF A COLUMN</h1>

In [19]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h1>CREATE NEW COLUMN</h1>

In [20]:
penguins['mass_kg'] = round(penguins['body_mass_g']/1000,1)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h1>EXPLORATORY DATA ANALYSIS (EDA)</h1>

<h3>1. Which Species has the heaviest body mass in kg?<h3>

In [21]:
Chinstrap_Pengu['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [22]:
Adelie_Pengu['body_mass_g'].mean()

np.float64(3700.662251655629)

In [23]:
Gentoo_Pengu['body_mass_g'].mean()

np.float64(5076.016260162602)

<h1>Insight</h1>

<h4>The heviest species on the presented data of Penguins in kilograms is GENTOO</h4>

<h1> 12 Questions for this Activity</h1>

In [24]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h3>1. Which Island on the presented data has the most numbers of female?</h3>

In [25]:
fem = penguins[penguins['sex'] == 'female']
fem_count = fem.groupby(['island','sex']).size()
print(fem_count)

island     sex   
Biscoe     female    80
           male       0
Dream      female    61
           male       0
Torgersen  female    24
           male       0
dtype: int64


In [26]:
total_fem = fem_count.sum()
print('Total Female Population: ',total_fem)

Total Female Population:  165


In [27]:
female_difference = (fem_count / fem_count.sum())*100
print('Distribution of the total Female Population within 3 islands\n',female_difference)

Distribution of the total Female Population within 3 islands
 island     sex   
Biscoe     female    48.484848
           male       0.000000
Dream      female    36.969697
           male       0.000000
Torgersen  female    14.545455
           male       0.000000
dtype: float64


<h2>Insights #1</h2>
<h5>The data presented above indicates that nearly <b>half</b> of all recorded female penguins can be found on the island of <b>Biscoe</b>.
</h5>
<p>Upon analysis, a noticeable margin in the female population can be observed based on the 3 islands they inhabit. With a total population of <b>165</b> female penguins, <b>48%</b> can be found in the island of Biscoe.
</p>
<p><i>The high number of females could indicate that Biscoe sustains <b>larger incubation areas</b> for different penguin species.</p>

<h3>2. Which Island on the presented data has the most numbers of male?</h3>

In [28]:
mal = penguins[penguins['sex']=='male']
mal_count = mal.groupby(['island','sex']).size()
print(mal_count)

island     sex   
Biscoe     female     0
           male      88
Dream      female     0
           male      63
Torgersen  female     0
           male      28
dtype: int64


In [29]:
total_mal = mal_count.sum()
print(f'Total Male Population: {total_mal}')

Total Male Population: 179


In [30]:
male_difference = (mal_count / mal_count.sum())*100
print('Distribution of the total Male Population within 3 islands\n',male_difference)

Distribution of the total Male Population within 3 islands
 island     sex   
Biscoe     female     0.000000
           male      49.162011
Dream      female     0.000000
           male      35.195531
Torgersen  female     0.000000
           male      15.642458
dtype: float64


<h2>Insights #2</h2>
<h5>The data presented above indicates that nearly <b>half</b> of all recorded male penguins can be found on the island of <b>Biscoe</b>.
</h5>
<p>Upon analysis, a noticeable margin in the male population can be observed based on the 3 islands they inhabit. With a total population of <b>179</b> male penguins, <b>49%</b> can be found in the island of Biscoe.
</p>
<p><i>Taking into account the previous female population within the same 3-island scope. It is safe to say that within the presented data, the island of <b>Biscoe has the densest penguin population.</p>

<h3>3. What is the overall Penguin population of the 3 islands combined?</h3>

<h2>Insights #3</h2>
<h5>The total penguin population of the 3 islands (Biscoe, Dream, Torgersen) is <b>344</b>.
</h5>
<p> Note that the number above is the sum of the male and female population. The dataset does not have enough information to indicate the number of mature penguins per species, island or year. Therefore, the fertility rate of the 3 islands, nor per island, cannot be determined.
</p>

In [31]:
population = (fem_count.sum() + mal_count.sum())
print(population)

344


<h4>4. What is the average bill length of all penguins presented?</h4>

In [32]:
adelie_bill = round(Adelie_Pengu['bill_length_mm'].mean())
chinstrap_bill = round(Chinstrap_Pengu['bill_length_mm'].mean())
gentoo_bill = round(Gentoo_Pengu['bill_length_mm'].mean())

print(f'The average bill of the Adelie penguin population is {adelie_bill}mm')
print(f'The average bill of the Chinstrap penguin population is {chinstrap_bill}mm')
print(f'The average bill of the Gentoo penguin population is {gentoo_bill}mm')



The average bill of the Adelie penguin population is 39mm
The average bill of the Chinstrap penguin population is 49mm
The average bill of the Gentoo penguin population is 48mm


In [33]:
round(penguins['bill_length_mm'].mean())

44

<h2>Insights #4</h2>
<h5>The average length of a penguin bill within the population of the 3 presented islands is <b>44mm</h5>
<p>Upon analysis, there is a noticeable difference among the species. Particularly the Adelie, having a bill that is shorter than the other two species. 

<i>The significance of the present difference of the Adelie species' bill length is subject for testing and is yet to be determined</p>

<h4>5. What is the average flipper length of all penguins presented?</h4>

In [34]:
adelie_flipper = round(Adelie_Pengu['flipper_length_mm'].mean())
chinstrap_flipper = round(Chinstrap_Pengu['flipper_length_mm'].mean())
gentoo_flipper = round(Gentoo_Pengu['flipper_length_mm'].mean())

print(f'The average flipper of the Adelie penguin population is {adelie_flipper}mm')
print(f'The average flipper of the Chinstrap penguin population is {chinstrap_flipper}mm')
print(f'The average flipper of the Gentoo penguin population is {gentoo_flipper}mm')

The average flipper of the Adelie penguin population is 190mm
The average flipper of the Chinstrap penguin population is 196mm
The average flipper of the Gentoo penguin population is 217mm


In [35]:
round(penguins['flipper_length_mm'].mean())

201

<h2>Insights #5</h2>
<h5>The average length of a penguin flipper within the population of the 3 presented island is <b>201mm</h5>
<p>Upon analysis, there is a noticeable difference among the species. Particularly the Gentoo, having a flipper that is longer than the other two species. 

<i>The significance of the present difference of the Gentoo species' flipper length is subject for testing and is yet to be determined</p>

<h4>6. What is the average bill depth of all penguins presented?</h4>

In [36]:
adelie_bill_depth = round(Adelie_Pengu['bill_depth_mm'].mean())
chinstrap_bill_depth = round(Chinstrap_Pengu['bill_depth_mm'].mean())
gentoo_bill_depth = round(Gentoo_Pengu['bill_depth_mm'].mean())

print(f'The average bill depth of the Adelie penguin population is {adelie_bill_depth}mm')
print(f'The average bill depth of the Chinstrap penguin population is {chinstrap_bill_depth}mm')
print(f'The average bill depth of the Gentoo penguin population is {gentoo_bill_depth}mm')

The average bill depth of the Adelie penguin population is 18mm
The average bill depth of the Chinstrap penguin population is 18mm
The average bill depth of the Gentoo penguin population is 15mm


In [37]:
round(penguins['bill_depth_mm'].mean())

17

<h2>Insights #6</h2>
<h5>The average depth of a penguin bill within the population of the 3 presented island is <b>17mm</h5>
<p>Upon analysis, there is a noticeable difference among the species. Particularly the Gentoo, having a bill depth that is shorter than the other two species. Interestingly, Adelie and Chinstrap shares the exact same bill depth of 18mm.

<i>The significance of the present difference of the Gentoo species' bill depth is subject for testing and is yet to be determined</p>

<h4>7. Is there a correlation between the sex and anatomical measurement of a penguin?</h4>

In [38]:
fem_bill_l = penguins[penguins['sex'] == 'female']['bill_length_mm']
mal_bill_l = penguins[penguins['sex'] == 'male']['bill_length_mm']

f_stat, p_value = stats.f_oneway(fem_bill_l, mal_bill_l)
print("F-stat:", f_stat, " P-value:", p_value)

F-stat: 39.65455984564767  P-value: 9.307501289283404e-10


In [39]:
fem_bill_d = penguins[penguins['sex'] == 'female']['bill_depth_mm']
mal_bill_d = penguins[penguins['sex'] == 'male']['bill_depth_mm']

f_stat, p_value = stats.f_oneway(fem_bill_d, mal_bill_d)
print("F-stat:", f_stat, " P-value:", p_value)

F-stat: 49.116852993833895  P-value: 1.2906481751545917e-11


In [40]:
fem_flipp_l = penguins[penguins['sex'] == 'female']['flipper_length_mm']
mal_flipp_l = penguins[penguins['sex'] == 'male']['flipper_length_mm']

f_stat, p_value = stats.f_oneway(fem_flipp_l, mal_flipp_l)
print("F-stat:", f_stat, " P-value:", p_value)

F-stat: 21.566658928918905  P-value: 4.88065474065626e-06


<h2>Insights #7</h2>
<h5>There is a large and <b>statistically significant correlation</b> between the Anatomical measurements of a penguin and it's sex</h5>
<p>With the 3 measurements present in the data set, it was found that male penguins tend to be <b>signifincantly bigger</b>. This aligns with the assumption that sex has influence to a penguins size.

<i>Important Note* The correlation was produced from the total population of Penguins. It is imporant to understand that Species can also be a huge variable in determining the influences on penguin size. This analysis only proves the big correlation in penguin sex and measurements, not including the potential size differences between species.</p>

<h4>8. In what year was the largest female penguin recorded for every species?</h4>

In [49]:
fem_year = fem.groupby(['year','species'])[['bill_length_mm','bill_depth_mm','flipper_length_mm','mass_kg']].max()
fem_year

bill_length_mm  bill_depth_mm  flipper_length_mm  mass_kg
year species                                                             
2007 Adelie               42.2           19.3              195.0      3.8
     Chinstrap            58.0           18.9              198.0      4.2
     Gentoo               48.7           14.6              215.0      5.0
2008 Adelie               40.9           19.0              202.0      3.8
     Chinstrap            50.5           18.4              200.0      3.9
     Gentoo               49.1           15.0              220.0      5.2
2009 Adelie               40.2           20.7              199.0      3.9
     Chinstrap            50.9           19.4              202.0      3.8
     Gentoo               50.5           15.5              222.0      5.2

<h2>Insights #8</h2>
<h5>The largest females recorded has varying large mearuements per year.</h5>
<p>Across 3 years of recording penguin measruements, it can be observed that these large recorded females have specific varying measurements. In 2007, the bill lengths of these species were of the largest within the 3-year scope. In 2008, the fllipper lengths were of the longest, and in 2009, the largest bill depth was recorded. All in the same 3-year scope and 3 species.
    
<i>What's more interesting is the results doesn't seem to repeat on a yearly basis. Meaning each penguin was not recorded twice.</p>

<h4>9. In what year was the largest male penguin recorded for every species?</h4>

In [48]:
mal_year = mal.groupby(['year','species'])[['bill_length_mm','bill_depth_mm','flipper_length_mm','mass_kg']].max()
mal_year

bill_length_mm  bill_depth_mm  flipper_length_mm  mass_kg
year species                                                             
2007 Adelie               46.0           21.5         200.915205      4.7
     Chinstrap            52.7           20.3         201.000000      4.4
     Gentoo               59.6           17.0         230.000000      6.3
2008 Adelie               45.8           21.1         208.000000      4.7
     Chinstrap            54.2           20.8         210.000000      4.8
     Gentoo               54.3           17.3         231.000000      6.0
2009 Adelie               45.6           20.5         210.000000      4.8
     Chinstrap            55.8           19.9         212.000000      4.4
     Gentoo               55.9           17.3         230.000000      6.0

<h2>Insights #9</h2>
<h5>The largest male was a Gentoo species recorded in 2007.</h5>
<p>Unlike the similar female analysis done previously, the males seem to have size similarities, almost repeated even in some species.
    
<i>Important Note* Which also applies to the previous analysis, the locations of these penguins are not determined here.</p>

<h4>10. What is the most common penguin species present in 2007?</h4>

In [43]:
penguins_2007 = penguins[penguins['year'] == 2007]

most_common_species = penguins_2007['species'].mode()[0]
print("Most common species in 2007:", most_common_species)


Most common species in 2007: Adelie


<h2>Insights #10</h2>
<h5>The most common species of penguin in 2007 is the Adelie.</h5>
<p>The results indicates that majority of the penguin population consist of adelies.
    
<i>It also worth mentioning the potential that Adelie has found a way to coexist with other penguin species in a very efficient way, having implications that they thrive on all 3 islands.</p>

<h4>11. What is the female-to-male ratio of the penguin population per species?</h4>

In [44]:
sex_count = penguins.groupby(['species','sex']).size().unstack()

sex_count['fem_to_male_ratio'] = sex_count['female'] / sex_count['male']
print(sex_count)


sex        female  male  fem_to_male_ratio
species                                   
Adelie         73    79           0.924051
Chinstrap      34    34           1.000000
Gentoo         58    66           0.878788


<h2>Insights #11</h2>
<h5>The species with the highest female-to-male ratio is the Chinstrap.</h5>
<p>Despite being the lowest in population, Chinstrap has the highest sex ratio of the 3 species.
    
<i>Adelie and Gentoo shares a more similar ratio but what they have in common is a lower female population.</p>

<h4>12. What species has the least amount of females?</h4>

In [45]:
least_females = penguins[penguins['sex'] == 'female'].groupby('species').size().idxmin()
print("Species with least females:", least_females)


Species with least females: Chinstrap


<h2>Insights #12</h2>
<h5>Least female population has been observed among the species of Chinstrap.</h5>
<p>The low female population post no big threat to their population, however, that is because they don't have a big population in the first place. In the span of 3 years, Chinstrap has the lowest penguin population among the 3 islands.
    
<i>This data can be used to understand if competition is involved in the low numbers on Chinstrap, is it just part of their biology to produce small number of offspring, or is it a mix of both?.</p>

<h1>TESTING CORRELATION</h1>

<h3>Pearson R Correlation</h3>

<hr><h4> Used for testing correlation between two <b>NUMERICAL VALUES</b></h4>

In [46]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])
print(weight_flipper_cor)

0.8712017673060115
